src/main/java/com/feedzai/commons/sql/abstraction/engine/impl/mysql/MySqlQueryExceptionHandler.java

```diff
@@ -29,8 +30,19 @@
  */
 public class MySqlQueryExceptionHandler extends QueryExceptionHandler {
 
+    /**
+     * The MySQL error code that indicates a unique constraint violation.
+     */
+    private static final int UNIQUE_CONSTRAINT_VIOLATION_ERROR_CODE = 1062;
+
     @Override
     public boolean isTimeoutException(final SQLException exception) {
         return exception instanceof MySQLTimeoutException || super.isTimeoutException(exception);
     }
+
+    @Override
+    public boolean isUniqueConstraintViolationException(final SQLException exception) {
+        return UNIQUE_CONSTRAINT_VIOLATION_ERROR_CODE
+                == exception.getErrorCode() || super.isUniqueConstraintViolationException(exception);
```

Comment:
Nitpicking: I'd rather keep both sides of the == expression on the same line,
and split the line by the `||` instead.
(same for Oracle, etc)